src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2363,7 +2388,8 @@ internal BoundStatement BindForOrUsingOrFixedDeclarations(VariableDeclarationSyn
             for (int i = 0; i < count; i++)
             {
                 var variableDeclarator = variables[i];
-                var declaration = BindVariableDeclaration(localKind, isVar, variableDeclarator, typeSyntax, declType, alias, diagnostics);
+                bool includeBoundType = i == 0; //To avoid duplicated expressions, only the first declaration should contain the bound type.
```

Comment:
>bool includeBoundType = i == 0; [](start = 16, length = 31)

We need to have a test similar to the one added in OutVarTests for all statements handled here (for, using block, using declaration, fixed), with the switch expression twist as well. Once you add those, I think you'll discover that we are missing some handling in LocalBinderFactory and in some BuildLocals implementations.

Reply:
By which you mean a test along the lines of
```
using(var int[M(out var y)] x)
```
?

Reply:
> By which you mean a test along the lines of
...

Yes

---
In reply to: [261491652](https://github.com/dotnet/roslyn/pull/32843#discussion_r261491652) [](ancestors = 261491652)